server/src/handlers/sql.rs

```diff
@@ -21,23 +22,39 @@ use crate::handlers::{
     prelude::*,
 };
 
+type ResponseRows = Vec<ResponseRow>;
+
 #[derive(Debug, Deserialize)]
 pub struct Request {
     query: String,
 }
 
-impl From<String> for Request {
-    fn from(query: String) -> Self {
-        Self { query }
-    }
-}
-
 // TODO(yingwen): Improve serialize performance
 #[derive(Serialize)]
 #[serde(rename_all = "snake_case")]
 pub enum Response {
     AffectedRows(usize),
-    Rows(Vec<HashMap<String, Datum>>),
+    #[serde(with = "http_format")]
```

Comment:
We can `impl Serialize for ResponseRow` and let `#[derive(Serialize)]` cover this. Then the sub-mod `http_format` and `row_format` can be removed. So do the util struct `Row`.